doc/api/diagnostics_channel.md

```diff
@@ -0,0 +1,282 @@
+# Diagnostics Channel
+
+<!--introduced_in=REPLACEME-->
+
+> Stability: 1 - Experimental
+
+<!-- source_link=lib/diagnostics_channel.js -->
+
+The `diagnostics_channel` module provides an API to create named channels
+to report arbitrary message data for diagnostics purposes.
+
+It can be accessed using:
+
+```js
+const diagnostics_channel = require('diagnostics_channel');
+```
+
+It is intended that a module writer wanting to report diagnostics messages
+will create one or many top-level channels to report messages through.
```

Comment:
why `top-level` channels? Are there any plans for other channels?

Reply:
Generating the channels at the top-level eliminates the publish-time lookups you'd have with a plain event emitter. It also allows for things like the prototype swap which makes channels near zero-cost when there are no subscribers.